Feature/refactor and type expectation diagnostic report

abegong opened the pull request:
WIP. Not yet ready for review.

abegong committed:
Initial scaffolding. One type and one test

abegong committed:
Scaffold most of the rest of the objects we'll need, with some tests.

abegong committed:
Add tests to unpack the behavior of @dataclass combined with SerializableDotDict

abegong committed:
More WIP, especially tests

abegong committed:
A spate of renaming and tightening

abegong committed:
Add SerializableDotDict.keys

abegong committed:
Update expectation_test_data_cases to use SerializableDictDot

abegong committed:
Fix gtypos

abegong committed:
Update SerializableDictDot, with tests

abegong committed:
The meat of the refactor, mostly done with some WIP

abegong committed:
Fixed test__count_positive_and_negative_example_cases

abegong committed:
Decruft

abegong committed:
Fix test_include_in_gallery_flag

abegong committed:
Make .to_dict recursive

abegong committed:
Linting

abegong committed:
Remove pydantic; Remove submessages in execution_engines; test_generate_diagnostic_checklist__first_iteration now passes

abegong committed:
Refactor _count_unexpected_cases_and_get_sub_messages into _count_unexpected_cases

abegong committed:
Fix docstring

abegong committed:
Clean up types and methods around LibraryMetadata

abegong committed:
Add support for enum serialization

abegong committed:
Implement LegacyAugmentedLibraryMetadataAdapter; Tidy up a little

abegong committed:
Small rename

abegong committed:
Remove pydantic

abegong committed:
Remove pydantic

abegong committed:
Remove pydantic

abegong committed:
Minor bugfixes. Many docstrings

abegong committed:
Add docstrings for test_data_cases

abegong committed:
Improve _get_metric_diagnostics_list

abegong committed:
Clean up _get_test_results

abegong committed:
Work on docstrings and more bug fixes. Re-up pydantic. Need to get out of the weeds soon.

abegong committed:
Deprecate LegacyAugmentedLibraryMetadataAdapter

abegong committed:
Tidy up rendering methods

abegong committed:
Add RendererTestDiagnostics object

abegong committed:
Add unit tests for _get_augmented_library_metadata

abegong committed:
Add unit tests for _get_augmented_library_metadata and _get_examples

abegong committed:
Unit test for _get_description_diagnostics

abegong committed:
Implement OtherExpectationTestDiagnostics

abegong committed:
Flow typed ExecutedExpectationTestCases through

abegong committed:
Remove dependence on global _registered_renderers. Get another test to pass.

abegong committed:
Get another test to pass

abegong committed:
Fix another test

abegong committed:
Fix another test

abegong committed:
Clean up tests and get business logic for Has at least one positive and negative example case, and all test cases pass working

abegong committed:
Decruft. Skip one failing test

abegong committed:
black

abegong committed:
isort

abegong committed:
Comment out pydantic

abegong committed:
isort and black

abegong committed:
Lint manually

abegong committed:
Lint manually

abegong committed:
Feature/changes from novuum jan3 review (#3941)

* Add ExpectationDiagnosticMaturityMessages class and refactor around it

* Add optional AugmentedLibraryMetadata.package and ExpectationDiagnosticCheckMessage.doc_url

abegong committed:
Decruft

abegong committed:
Merge branch 'hackathon-docs' into feature/refactor_and_type_expectation_diagnostic_report

abegong committed:
Linting

abegong committed:
Merge branch 'feature/refactor_and_type_expectation_diagnostic_report' of github.com:great-expectations/great_expectations into feature/refactor_and_type_expectation_diagnostic_report

abegong committed:
Fix mismatched string in tests

donaldheppner commented on great_expectations/core/expectation_diagnostics/expectation_diagnostics.py:
```diff
@@ -0,0 +1,251 @@
+from great_expectations.core.expectation_diagnostics.expectation_test_data_cases import (
+    ExpectationTestDataCases,
+)
+from great_expectations.core.expectation_diagnostics.supporting_types import (
```
Remove the supporting_types package

donaldheppner commented on great_expectations/core/expectation_diagnostics/expectation_diagnostics.py:
```diff
@@ -0,0 +1,251 @@
+    ExpectationRendererDiagnostics,
+    ExpectationTestDiagnostics,
+)
+from great_expectations.types import SerializableDictDot
```
modules should be semantic, not reflecting the nature of the code in the module

donaldheppner commented on great_expectations/core/expectation_diagnostics/expectation_diagnostics.py:
```diff
@@ -0,0 +1,251 @@
+        """Check whether the Expectation has a library_metadata object"""
+
+        return ExpectationDiagnosticCheckMessage(
+            **{
```
use dot notation

donaldheppner commented on great_expectations/core/expectation_diagnostics/expectation_diagnostics.py:
```diff
@@ -0,0 +1,251 @@
+        if short_description not in {"", "\n", "TODO: Add a docstring here", None}:
+            return ExpectationDiagnosticCheckMessage(
+                **{
+                    "message": message,
```
use dot notation

donaldheppner commented on tests/expectations/test_expectation_diagnostics.py:
```diff
@@ -0,0 +1,430 @@
+
+
+def test__check_example_cases__with_enough_test_cases_and_no_failing():
+    print(expectation_test_data_case)
```
please remove

abegong replied:
```suggestion
```

kenwade4 committed:
Merge remote-tracking branch 'origin/develop' into feature/refactor_and_type_expectation_diagnostic_report

kenwade4 committed:
Delete test_expectation__get_execution_engine_dict

_get_execution_engine_dict was replaced by
_get_execution_engine_diagnostics

abegong committed:
Merge branch 'hackathon-docs' into feature/refactor_and_type_expectation_diagnostic_report

abegong committed:
Update tests/expectations/test_expectation_diagnostics.py

Co-authored-by: Don Heppner <donald.heppner@gmail.com>

kenwade4 committed:
Merge remote-tracking branch 'origin/hackathon-docs' into feature/refactor_and_type_expectation_diagnostic_report

kenwade4 committed:
Implement ExpectationDiagnostics.to_json_dict method and use in build_gallery.py script

kenwade4 committed:
Update build_gallery.py script to collect expectation tracebacks and display before writing gallery json file

kenwade4 committed:
Allow ExpectationTestDataCases to receive 'schemas'

kenwade4 committed:
Add commented-out 'test_backends' attribute to ExpectationTestDataCases class

kenwade4 committed:
Add a mysql docker image that works with M1 Macs

% docker-compose up
Pulling mysql_db (mysql:8.0.20)...
8.0.20: Pulling from library/mysql
ERROR: no matching manifest for linux/arm64/v8 in the manifest list entries

kenwade4 committed:
Ensure that the 'configuration' object passed to renderers are typed objects

kenwade4 committed:
Update build_gallery.py script to not die if there are expectation tracebacks

kenwade4 committed:
Use dot notation for args passed to ExpectationDiagnosticCheckMessage
